### ISSUE TITLE
restore original disable-physics functionality

### ISSUE THREAD

**Issue by Folas1337** [TehBrian/BuildersUtilities#63]
I know about #11 but thought I'd open this issue anyway.

I had a user tell me that since I have set `disable-physics: true`, their fences no longer connect properly, providing this video: 
https://github.com/TehBrian/BuildersUtilities/assets/39222968/b1982dc3-1d6e-42c1-8640-59541b1c481b

I also verified this on a clean server with no other plugins than BU and enabling disable-physics.

**Comment by TehBrian** [comment]
Yeah, that's unfortunately expected behavior when `disable-physics` is enabled in its current state.

The way to fix this would be either: to provide more fine-grained config settings for disabling physics, or to not disable some specific events such as fences connecting.

Fine-grained config settings sound nice in theory, but since there are so many physics events in the game, the amount of options that the plugin could theoretically provide could grow way too large and become annoying for server owners to sift through. Additionally, Bukkit doesn't provide all physics events to plugins, so it's not even like we can definitively provide options for *everything*.

On the other hand, adding sensible defaults like allowing fences to connect sounds reasonable, but it's tough to know what server owners will actually end up wanting. It's hard to make everyone happy.

I'll look more into this later. If you have thoughts/ideas, please do share.

**Comment by Folas1337** [comment]
It seems to be handled the way "my builders" want it in the original plugin (Arcaniax's). What exactly is the problem in doing it that way or why did you change it to begin with? :D Maybe that would help me try and understand why you changed it and the whole situation...

Also fine-grained config settings sound like a pain, although it gives a lot of control, I absolutely get how this would overwhelm owners and maintaining them all through all the updates in the future is also going to be painful, I'm not exactly for that idea.

**Comment by TehBrian** [comment]
> What exactly is the problem in doing it that way or why did you change it to begin with?

Arcaniax's plugin chose to provide sensible defaults. (You can see the logic that it uses [here](https://github.com/Arcaniax-Development/Builders-Utilities/blob/master/src/main/java/net/arcaniax/buildersutilities/listeners/BlockPhysicsListener.java).) I don't really remember why I removed all of that from this fork, but I think it was because I didn't personally use those features at the time. (This fork was originally intended for my personal use on my Minecraft server, so I was fine with tossing things I didn't need.)

You bring up a good point that the original plugin worked fine, though, and since that functionality is at least better than the current functionality this fork has, I'll just go ahead and bring back the original logic for physics events.

**Comment by TehBrian** [comment]
It is interesting to note, though, that the same fence-related issue that occurs in the video that you sent also exists in the original BuildersUtilities plugin. https://github.com/Arcaniax-Development/Builders-Utilities/issues/53

I suspect that unless I modify the original plugin's logic a bit, that issue will continue to happen.

**Comment by Folas1337** [comment]
Until we find a basically perfect solution, I still suggest going back to the old way things are/were.
I don't mind trying other things too but if we can't really improve on the original one, why change it? At least that's what I'm thinking :)

**Comment by TehBrian** [comment]
Oh, I totally agree. The original logic is already in the plugin as of https://github.com/TehBrian/BuildersUtilities/commit/b1ba84e26f521fca6e57c68bd211090c2c971f2f. Now, I'm currently working on trying to improve it. :)

**Comment by Folas1337** [comment]
Not sure if it's rude to ask if it's tough to "trying to improve it" or if I should just give you a little hint that I'm still kiiinda waiting for this :D 
I really mean no offense, just thought I'd bring it up again after this much time :)

**Comment by TehBrian** [comment]
You're totally fine. Sorry for the long wait on this! I've been meaning to get a release out for a while but have been quite busy with other stuff. I'll publish a new ASAP. :)

**Comment by Folas1337** [comment]
Heyyyyy, quick reminder for this in case it's been forgotten about. If you didn't forget, I was never here :)

**Comment by TehBrian** [comment]
> Heyyyyy, quick reminder for this in case it's been forgotten about. If you didn't forget, I was never here :)

Not forgotten! I'm getting around to it. Thanks for your patience <3

**Comment by Folas1337** [comment]
Works as badly as ever, makes my users happy apparently, so I'll close this issue :D 
Thanks for the release and I hope you're having a good day :)

**Comment by TehBrian** [comment]
You're welcome! If you ever come up with any ideas to improve the physics options, do feel free to create a new issue. Have a good day as well. :)

**Comment by TehBrian** [comment]
YAY FINALLY FIXED FOR REALZIES

for future generations looking upon this archive, the summary is that i added an air check that did not need to be there and prevented the physics listener from canceling relevant events
